src/main/java/org/springframework/data/redis/connection/lettuce/LettuceConnectionFactory.java

```diff
@@ -979,6 +980,9 @@ public void stop() {
 			dispose(reactiveConnectionProvider);
 			reactiveConnectionProvider = null;
 
+			dispose(clusterCommandExecutor);
```

Comment:
In these changes, `ClusterCommandExecutor` is being disposed after disposing the connection provider and so there's no behavioral change. Shouldn't `clusterCommandExecutor` be disposed (and thus its shared connection returned to the pool) before disposing `connectionProvider`?